FlashDevelop/Dialogs/ClipboardHistoryDialog.cs

```diff
@@ -0,0 +1,310 @@
+            {
+                listBox.Items.Insert(0, data);
+            }
+            listBox.EndUpdate();
+
+            btnClear.Enabled = listBox.Items.Count > 0;
+        }
+
+        #endregion
+
+        #region Events
+
+        private void ListBox_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            selectedData = (ClipboardTextData) listBox.SelectedItem;
+            ShowContent();
+            UpdateButtons();
+        }
+
+        private void BtnClear_Click(object sender, EventArgs e)
+        {
+            listBox.SelectedIndex = -1;
+
+            ClipboardManager.History.Clear();
+            listBox.Items.Clear();
+            btnClear.Enabled = false;
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Opens a <see cref="ClipboardHistoryDialog"/> and get the user selected <see cref="ClipboardTextData"/>.
+        /// </summary>
+        /// <param name="data">User selected <see cref="ClipboardTextData"/>.</param>
+        public static bool Show(out ClipboardTextData data)
+        {
+            try
+            {
+                current = new ClipboardHistoryDialog();
+                Globals.MainForm.ThemeControls(current);
+                current.SelectFirstItem();
+                var dialogResult = current.ShowDialog(Globals.MainForm);
+                data = current.SelectedData;
+                return dialogResult == DialogResult.OK;
+            }
+            finally
+            {
+                current?.Dispose();
+                current = null;
+            }
+        }
+
+        /// <summary>
+        /// Updates the clipboard history list by adding the new clipboard data to the list.
+        /// </summary>
+        public static void UpdateHistory()
+        {
+            if (current != null)
+            {
+                current.AddNewClipboardData();
+            }
+        }
+
+        private void SelectFirstItem()
+        {
+            if (listBox.Items.Count > 0)
+            {
+                listBox.SelectedIndex = 0;
+            }
+        }
+
+        private void AddNewClipboardData()
+        {
+            listBox.BeginUpdate();
+            if (listBox.Items.Count == ClipboardManager.History.Count)
+            {
+                listBox.Items.RemoveAt(listBox.Items.Count - 1);
+            }
+            listBox.Items.Insert(0, ClipboardManager.History.PeekEnd());
+            listBox.EndUpdate();
+
+            btnClear.Enabled = listBox.Items.Count > 0;
+        }
+
+        private void ShowContent()
+        {
+            richTextBox.Clear();
+
+            if (selectedData != null)
+            {
+                if (selectedData.Format == DataFormats.Text)
+                {
+                    richTextBox.Text = selectedData.Text;
+                }
+                else if (selectedData.Format == DataFormats.Rtf)
+                {
+                    richTextBox.Rtf = selectedData.Rtf;
```

Comment:
If I understood correctly you are not copying RTF anymore, so I guess this if is not needed.